src/inc/XmlWriter.hpp

```diff
@@ -0,0 +1,44 @@
+//
+//  Copyright (C) 2019 Microsoft.  All rights reserved.
+//  See LICENSE file in the project root for full license information.
+// 
+#pragma once
+
+#include "ComHelper.hpp"
+
+#include <stack>
+#include <string>
+
+namespace MSIX {
+
+    // common attribute names
+    static const char* xmlnsAttribute = "xmlns";
+
+    // This is a super light xml writer that doesn't use any xml libraries and 
+    // just writes to a stream the basics of an xml file.
+    class XmlWriter final
+    {
+    public:
+        typedef enum
+        {
+            OpenElement = 1,
+            ClosedElement = 2,
+            Finish = 3
+        }
+        State;
+
+        XmlWriter(const std::string& root);

```

Comment:
In the future, we may want to be able to specify the target stream.

Reply:
Would it take ownership of the stream? I made a constructor that takes a ComPtr<IStream>&, if we decide that it shouldn't take ownership of it then I'll change it to IStream* and add ref it

---
In reply to: [278233393](https://github.com/Microsoft/msix-packaging/pull/95#discussion_r278233393) [](ancestors = 278233393)

Reply:
Was more of a thought that a required change, but no, it shouldn't take ownership of the stream.

---
In reply to: [278361740](https://github.com/Microsoft/msix-packaging/pull/95#discussion_r278361740) [](ancestors = 278361740,278233393)